apps/indexer/src/api/controller/governance-activity/controller.ts

```diff
@@ -132,19 +123,23 @@ export function governanceActivity(
       },
     }),
     async (context) => {
-      const { daoId } = context.req.valid("param");
       const { days } = context.req.valid("query");
 
-      const data = await repository.getProposalsCompare(daoId, days);
+      const data = await repository.getProposalsCompare(days);
       if (!data) {
         return context.json({ error: "No data found" }, 404);
       }
       const changeRate =
-        data.oldProposalsLaunched === 0
-          ? 0
-          : data.currentProposalsLaunched / data.oldProposalsLaunched - 1;
+        data.oldProposalsLaunched &&
+        data.currentProposalsLaunched / data.oldProposalsLaunched - 1;
```

Comment:
What if oldProposalsLaunched <= 1  ? Then the result will be infinity heheheheh

Reply:
the calculation here is `data.currentProposalsLaunched / data.oldProposalsLaunched` and then the result is subtracted -1 to become a ratio. example

| Current | Old | current/old (ratio) | current/old - 1 (% change) | Meaning |
|---------|-----|----------------------|------------------------------|---------|
| 100 | 100 | 1.0 | 0.0 | No change (0%) |
| 150 | 100 | 1.5 | 0.5 | 50% increase |
| 200 | 100 | 2.0 | 1.0 | 100% increase |
| 50 | 100 | 0.5 | -0.5 | 50% decrease |
| 0 | 100 | 0.0 | -1.0 | 100% decrease |


The -1 converts:
Ratio of 1.0 → Change of 0% (no change)
Ratio of 1.5 → Change of +50% (50% increase)
Ratio of 0.5 → Change of -50% (50% decrease)